src/jdk.crypto.cryptoki/share/native/libj2pkcs11/p11_objmgmt.c

```diff
@@ -245,13 +247,35 @@ JNIEXPORT void JNICALL Java_sun_security_pkcs11_wrapper_PKCS11_C_1GetAttributeVa
     }
 
     rv = (*ckpFunctions->C_GetAttributeValue)(ckSessionHandle, ckObjectHandle, ckpAttributes, ckAttributesLength);
-    if (ckAssertReturnValueOK(env, rv) != CK_ASSERT_OK) {
+
+    if (rv != CKR_OK) {
+        if (rv == CKR_ATTRIBUTE_SENSITIVE || rv == CKR_ATTRIBUTE_TYPE_INVALID) {
+            msg = malloc(80); // should be more than sufficient
+            if (msg == NULL) {
+                throwOutOfMemoryError(env, 0);
+                free(ckpAttributes);
+                return;
+            }
+            // format msg w/ attribute(s) whose value is unavailable
+            temp1 = msg;
+            temp2 = msg + 80;
+            for (i = 0; i < ckAttributesLength && temp1 < temp2; i++) {
```

Comment:
I think that this loop will append at most 11 bytes to the string each time (is this correct?), if so, we can make the check `temp1 < temp2 - 12` to count the final null and avoid running off the end with a buffer overflow.

Reply:
I apologize. This counting code is correct and doesn't have any of the issues I claimed. `snprintf` takes care of everything and I just missed it last night.

Reply:
It's alright, thanks for checking and review.